tests/handlers/records-write.spec.ts

```diff
@@ -3308,6 +3308,146 @@ export function testRecordsWriteHandler(): void {
           expect(recordsReadReply.entries?.length).to.equal(2);
         });
 
+        it('should allow authorization if protocol message size is within min and max size', async () => {
+          const alice = await TestDataGenerator.generatePersona();
+          TestStubGenerator.stubDidResolver(didResolver, [alice]);
+
+          const protocolDefinition = {
+            protocol  : 'http://blob-size.xyz',
+            published : true,
+            types     : {
+              blob: {}
+            },
+            structure: {
+              blob: {
+                $size: {
+                  min : 1,
+                  max : 1000
+                }
+              }
+            }
+          };
+
+          const protocol = protocolDefinition.protocol;
+          const protocolConfig = await TestDataGenerator.generateProtocolsConfigure({
+            author: alice,
+            protocolDefinition,
+          });
+
+          const protocolConfigureReply = await dwn.processMessage(alice.did, protocolConfig.message);
+          expect(protocolConfigureReply.status.code).to.equal(202);
+
+          // test min record size
+          const data = TestDataGenerator.randomBytes(1);
+          const testRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : alice,
+            recipient    : alice.did,
+            protocol,
+            protocolPath : 'blob',
+            data
+          });
+
+          const reply = await dwn.processMessage(alice.did, testRecord.message, { dataStream: testRecord.dataStream });
+          expect(reply.status.code).to.equal(202);
+
+          // test max record size
+          const data2 = TestDataGenerator.randomBytes(1000);
+          const testRecord2 = await TestDataGenerator.generateRecordsWrite({
+            author       : alice,
+            recipient    : alice.did,
+            protocol,
+            protocolPath : 'blob',
+            data         : data2
+          });
+
+          const reply2 = await dwn.processMessage(alice.did, testRecord2.message, { dataStream: testRecord2.dataStream });
+          expect(reply2.status.code).to.equal(202);
```

Comment:
Super small nit if you don't mind adding the negative test here, so a failure.
I know the failure is tested elsewhere but we like to include negative tests in the same place where positives are tested in most places.

Same comments for the other 2 below.